### PR TITLE
test: harden resilience verification

### PR DESCRIPTION
**Pull request by thomhurst**
## Summary

- add 29 deterministic tests across core fluent APIs, partition lifecycle, analyzers, testing helpers, Chaos boundaries, adapters, and real integration paths
- add allocation guards for outcome-state execution, dynamic timeout, and void fallback paths
- fix modern gRPC per-write cancellation by shipping a `netstandard2.1` adapter asset while retaining `netstandard2.0` and `net10.0`
- exercise the retained `netstandard2.0` and modern `netstandard2.1` gRPC assets in separate runtime consumer projects
- enforce project-specific test discovery floors, automatic test-suite registration, 94% line / 89% branch coverage, and a 72% mutation score floor
- broaden model, mutation, stress, and benchmark workflow triggers for relevant configuration changes

## Coverage

- line coverage: 94.94% (previously 93.76%)
- branch coverage: 90.45% (previously 89.79%)
- CI floor: 94% lines / 89% branches

## Validation

- `dotnet build Kevlar.slnx -c Release`
- all 9 executable suites: 1,146 passed, 0 failed, 0 skipped
- `pwsh .github/scripts/Assert-TestSuiteRegistration.ps1`
- merged Cobertura coverage gate
- allocation gates
- package layout, runtime consumers, trimmed and single-file publish checks
- 118 compiled documentation snippets and documented behavior execution
- `npm run build` in `docs/`
- 30-second stress smoke test (Kevlar: 0 B/op)
- workflow YAML, Stryker JSON, and test-suite registration validation

NativeAOT package verification remains covered by Linux CI; local Windows validation reports it unsupported as designed. The full Stryker gate runs in CI; focused contract assertions target 14 survivors from the prior 71.21% report.

### CODE REVIEW

File: tests/Kevlar.NetStandard.Tests/Kevlar.NetStandard.Tests.csproj [hunk 1]
```diff
@@ -17,7 +17,7 @@
     <ProjectReference Include="..\..\src\Kevlar.Extensions.Http\Kevlar.Extensions.Http.csproj"
                       SetTargetFramework="TargetFramework=netstandard2.0" />
     <ProjectReference Include="..\..\src\Kevlar.Extensions.Grpc\Kevlar.Extensions.Grpc.csproj"
-                      SetTargetFramework="TargetFramework=netstandard2.0" />
+                      SetTargetFramework="TargetFramework=netstandard2.1" />
```
[the remark this answers is not in the document]
Comment (thomhurst) [reply tests/Kevlar.NetStandard.Tests/Kevlar.NetStandard.Tests.csproj]: Fixed in 1ebfbbe. The original compatibility project again consumes the netstandard2.0 gRPC asset and retains its blocked-write lifetime-cancellation tests. A separate Kevlar.NetStandard21.Tests project now consumes netstandard2.1 and verifies per-write cancellation, with both projects registered in the solution, CI, coverage, and docs.
